app.py

```diff
@@ -162,17 +177,61 @@ def show_weights(self):
         if len(self.weights) == 0:
             return
         if self.show_weight.get():
-            for weight in self.weights:
-                x, y, value = weight
+            for line in self.canvas_obj[self.OBJ_TYPE.LINE].values():
+                x_start,y_start = self.coordinates[line['point_start']]
+                x_end, y_end = self.coordinates[line['point_end']]
+                x, y = self.medium_coordinates(x_start,y_start,x_end, y_end)
+                value = line['weight']
                 r = int(self.r / 2) * len(str(value))
-                self.ids.append(self.canvas.create_oval(x - r, y - r, x + r, y + r, fill=self.BG, width=0))
-                self.ids.append(self.canvas.create_text(x, y, text=value, font="{} {}".format(self.FONT, str(r))))
+                weight_id = self.canvas.create_oval(x - r, y - r, x + r, y + r, fill=self.BG, width=0)
+                text_id = self.canvas.create_text(x, y, text=value, font="{} {}".format(self.FONT, str(r)))
+                self.canvas_obj[self.OBJ_TYPE.WEIGHT][weight_id] = {'text_id':text_id}
```

Comment:
I suppose you can store oval_id and text_id of some weight as a line attribute. See comment above. And also code style here:
{'text_id':text_id} - not cool;
{'text_id': text_id} - better